Files/DataModels/SidebarPinnedModel.cs

```diff
@@ -338,51 +364,60 @@ public async Task AddAllItemsToSidebar()
-            };
-            favoriteSection = new LocationItem()
+            if (!App.AppSettings.ShowFavoritesSection)
             {
-                Text = "SidebarFavorites".GetLocalized(),
-                Section = SectionType.Favorites,
-                SelectsOnInvoked = false,
-                Icon = UIHelpers.GetImageForIconOrNull(IconResources?.FirstOrDefault(x => x.Index == Constants.Shell32.QuickAccess).Image),
-                Font = MainViewModel.FontName,
-                ChildItems = new ObservableCollection<INavigationControlItem>()
-            };
-            try
+                SidebarControl.SideBarItemsSemaphore.Release();
+            }
+            else
             {
-                if (homeSection != null)
+                try
                 {
-                    AddItemToSidebarAsync(homeSection);
-                }
+                    homeSection = new LocationItem()
+                    {
+                        Text = "SidebarHome".GetLocalized(),
+                        Section = SectionType.Home,
+                        Font = MainViewModel.FontName,
+                        IsDefaultLocation = true,
+                        Icon = await CoreApplication.MainView.DispatcherQueue.EnqueueAsync(() => new BitmapImage(new Uri("ms-appx:///Assets/FluentIcons/Home.png"))),
+                        Path = "Home".GetLocalized(),
+                        ChildItems = new ObservableCollection<INavigationControlItem>()
+                    };
+                    favoriteSection = new LocationItem()
+                    {
+                        Text = "SidebarFavorites".GetLocalized(),
+                        Section = SectionType.Favorites,
+                        SelectsOnInvoked = false,
+                        Icon = UIHelpers.GetImageForIconOrNull(IconResources?.FirstOrDefault(x => x.Index == Constants.Shell32.QuickAccess).Image),
+                        Font = MainViewModel.FontName,
+                        ChildItems = new ObservableCollection<INavigationControlItem>()
+                    };
 
-                for (int i = 0; i < FavoriteItems.Count(); i++)
-                {
-                    string path = FavoriteItems[i];
-                    await AddItemToSidebarAsync(path);
-                }
+                    if (homeSection != null)
+                    {
+                        AddItemToSidebarAsync(homeSection);
+                    }
 
-                if (!SidebarControl.SideBarItems.Contains(favoriteSection))
+                    for (int i = 0; i < FavoriteItems.Count(); i++)
+                    {
+                        string path = FavoriteItems[i];
+                        await AddItemToSidebarAsync(path);
+                    }
+
+                    if (!SidebarControl.SideBarItems.Contains(favoriteSection))
+                    {
+                        SidebarControl.SideBarItems.BeginBulkOperation();
+                        var index = 0; // First section
```

Comment:
I added this line to keep consistency, but this could be deleted and just go with:
 SidebarControl.SideBarItems.Insert(0, favoriteSection);